stl/inc/ranges

```diff
@@ -1913,6 +1913,50 @@ namespace ranges {
         };
 
         inline constexpr _Drop_fn drop;
+
+        // VARIABLE views::counted
+        class _Counted_fn {
+        private:
+            enum class _St { _Span, _Subrange, _Subrange_counted };
+
+            template <class _It>
+            _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
+                if constexpr (contiguous_iterator<_It>) {
+                    return {_St::_Span, noexcept(span{_STD to_address(_STD declval<_It>()), iter_difference_t<_It>{}})};
+                } else if constexpr (random_access_iterator<_It>) {
+                    return {_St::_Subrange,
+                        noexcept(subrange{_STD declval<_It>(), _STD declval<_It>() + iter_difference_t<_It>{}})};
+                } else {
+                    return {_St::_Subrange_counted,
+                        noexcept(subrange{
+                            counted_iterator{_STD declval<_It>(), iter_difference_t<_It>{}}, default_sentinel})};
+                }
+            }
+
+            template <class _It>
+            static constexpr _Choice_t<_St> _Choice = _Choose<_It>();
+
+        public:
+            // clang-format off
+            template <input_or_output_iterator _It, class _Diff>
+                requires convertible_to<_Diff, iter_difference_t<_It>>
+            _NODISCARD constexpr auto operator()(_It _First, const _Diff _Count) const

```

Comment:
This (1) introduces an additional (user-observable) move for the iterator, and (2) requires the second argument to be copy constructible which the working draft does not. I suggest taking the first parameter by forwarding reference, and changing the type of the second parameter to the difference type of the first to force the conversion onto the caller. i.e., something like:
```suggestion
            template <class _It>
                requires input_or_output_iterator<decay_t<_It>>
            _NODISCARD constexpr auto operator()(
                _It&& _First, const iter_difference_t<decay_t<_It>> _Count) const
```
with fixups elsewhere to replace uses of `_It` with `decay_t<_It>` and perfect forward, etc.